### DISCUSSION TITLE
Help to hack `switches` 🙏

### DISCUSSION THREAD

**Discussion by Cicorione** [speedata/publisher#626] [Q&A]
Hi @pgundlach,

I already tested that **you can't break tags** with the `switch` one, but I wonder if there is any tricky work-around for hacking them.

I have switch case blocks that now I would like to refactor because are "long" while they actually change for few details, for instance:

![image](https://github.com/user-attachments/assets/f0dc8dd6-6d53-4f99-986d-5ed08dae0f69)

`left`
```xml
<PlaceObject column="1">
<Table width="{$noc * 0.75}" stretch="no" fontfamily="title" border-collapse="collapse">
<Columns>
<Column width="4mm" />
<Column width="3*" />
[...]
```

`right`
```xml
<PlaceObject column="1" row="10">
<Table width="8" stretch="no" fontfamily="title" border-collapse="collapse">
<Columns>
<Column width="4mm" />
<Column width="4*" />
[...]
```

The goal here is to reduce code and to improve its understanding; any recommendations? 

Thanks! 🙏

**Comment by pgundlach** [comment]
Perhaps you need something like a function which accepts a few parameters and then you can modify the code that is inserted? See  https://doc.speedata.de/publisher/en/programming/#ch-programmingfunctions

Or I completely misunderstood your question.

**Reply by Cicorione** [reply]
Thank you Patrick, great advise! 🙏

I have to study it and see what I am able to do.

💪
